### ISSUE TITLE
Transformation of org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener fails with NPE

### ISSUE THREAD

**Issue by jluehe** [eclipse/transformer#544]
From the thousands of classes we transform, there is one that fails consistently with an NPE: `org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener`. 

I have attached an executable jar file which reproduces the issue: `java -jar eclipse-transformer-npe.jar` 

I have also attached (`eclipse-transformer-npe.zip`) the maven project containing the source code of my reproducer test:
```
import com.google.common.collect.Maps;
import org.apache.commons.io.IOUtils;
import org.eclipse.transformer.action.ActionContext;
import org.eclipse.transformer.action.ByteData;
import org.eclipse.transformer.action.impl.ActionContextImpl;
import org.eclipse.transformer.action.impl.ByteDataImpl;
import org.eclipse.transformer.action.impl.ClassActionImpl;
import org.eclipse.transformer.action.impl.SelectionRuleImpl;
import org.eclipse.transformer.action.impl.SignatureRuleImpl;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;
import java.io.InputStream;
import java.nio.ByteBuffer;
import java.util.Collections;
import java.util.Properties;

/**
 * Test class which reproduces NullPointerException during transformation of class
 * <code>org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener</code>
 *
 * @author jluehe 
 */
public class MyTest {

    static {
        System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", "trace");
    }

    public static void main(String[] args) throws Exception {
        final Properties packageRenames = new Properties();
        try (InputStream is = MyTest.class.getClassLoader().getResourceAsStream("org/eclipse/transformer/jakarta/jakarta-renames.properties")) {
            packageRenames.load(is);
        }
        final Logger logger = LoggerFactory.getLogger(MyTest.class.getName());
        final ActionContext actionContext = new ActionContextImpl(
                logger,
                new SelectionRuleImpl(logger, Collections.emptyMap(), Collections.emptyMap()),
                new SignatureRuleImpl(logger, Maps.fromProperties(packageRenames), null, null, null, null, null,
                        Collections.emptyMap()));
        final MyTest myTest = new MyTest();
        myTest.transform("org.eclipse.jetty.servlets.HeaderFilter", actionContext); // Success
        myTest.transform("org.eclipse.jetty.servlets.DoSFilter", actionContext); // Success
        myTest.transform("org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener", actionContext); // Failure
    }

    private void transform(String name, ActionContext actionContext) throws IOException {
        final byte[] classbytes;
        try (InputStream is = getClass().getClassLoader().getResourceAsStream(name.replace('.', '/').concat(".class"))) {
            classbytes = IOUtils.toByteArray(is);
        }
        final ClassActionImpl classAction = new ClassActionImpl(actionContext);
        System.out.println("Transforming class bytes for class: " + name);
        final ByteData byteData = classAction.apply(new ByteDataImpl(name, ByteBuffer.wrap(classbytes), null));
        final byte[] ret = byteData.buffer().array();
        if (classbytes != ret) {
            System.out.println("Transformed class bytes for class: " + name);
        }
    }
}
```
Logging output:
```
...
[main] DEBUG MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] as [ org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener ] 
[main] TRACE MyTest -   Super [ org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener ]
[main] TRACE MyTest -   Interfaces [ 0 ]
[main] TRACE MyTest -   Fields [ 1 ]
[main] TRACE MyTest -     [ this$0 ] [ Lorg/eclipse/jetty/servlets/DoSFilter; ]
[main] TRACE MyTest -   Methods [ 2 ]
[main] TRACE MyTest -     [ <init> ] [ (Lorg/eclipse/jetty/servlets/DoSFilter;)V ]
[main] TRACE MyTest -     [ onTimeout ] [ (Ljavax/servlet/AsyncEvent;)V ]
[main] TRACE MyTest - class org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener
[main] TRACE MyTest -   extends org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener
[main] DEBUG MyTest - Member onTimeout.(Ljavax/servlet/AsyncEvent;)V > (Ljakarta/servlet/AsyncEvent;)V
[main] DEBUG MyTest - Attribute [ 1 ] [ LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljavax/servlet/AsyncEvent;:1] ] -> [ LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljakarta/servlet/AsyncEvent;:1] ]
[main] DEBUG MyTest - Attribute [ 0 ] [ Code (23 bytes) [LineNumberTable [0:1306, 17:1307, 22:1308], LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljavax/servlet/AsyncEvent;:1]] ] -> [ Code (23 bytes) [LineNumberTable [0:1306, 17:1307, 22:1308], LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljakarta/servlet/AsyncEvent;:1]] ]
[main] DEBUG MyTest - Method public onTimeout (Ljavax/servlet/AsyncEvent;)V [Code (23 bytes) [LineNumberTable [0:1306, 17:1307, 22:1308], LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljavax/servlet/AsyncEvent;:1]], Exceptions [java/io/IOException]] -> public onTimeout (Ljakarta/servlet/AsyncEvent;)V [Code (23 bytes) [LineNumberTable [0:1306, 17:1307, 22:1308], LocalVariableTable [0:23:this:Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener;:0, 0:23:event:Ljakarta/servlet/AsyncEvent;:1]], Exceptions [java/io/IOException]]
[main] DEBUG MyTest -   Constant pool: 53
[main] TRACE MyTest - Constant [   1 ] [                9 ] [ FieldrefInfo:2:3 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [   2 ] [                7 ] [ ClassInfo:4 ]
[main] DEBUG MyTest - Skip class org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener (unchanged)
[main] TRACE MyTest - Constant [   3 ] [               12 ] [ NameAndTypeInfo:5:6 ]
[main] TRACE MyTest - Skip name-and-type Lorg/eclipse/jetty/servlets/DoSFilter; (unchanged)
[main] TRACE MyTest - Constant [   4 ] [                1 ] [ org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener (unchanged)
[main] TRACE MyTest - Skip UTF8 org/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener (unchanged)
[main] TRACE MyTest - Constant [   5 ] [                1 ] [ this$0 ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: this$0 (unchanged)
[main] TRACE MyTest - Skip UTF8 this$0 (unchanged)
[main] TRACE MyTest - Constant [   6 ] [                1 ] [ Lorg/eclipse/jetty/servlets/DoSFilter; ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Lorg/eclipse/jetty/servlets/DoSFilter; (unchanged)
[main] TRACE MyTest - Skip UTF8 Lorg/eclipse/jetty/servlets/DoSFilter; (unchanged)
[main] TRACE MyTest - Constant [   7 ] [               10 ] [ MethodrefInfo:8:9 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [   8 ] [                7 ] [ ClassInfo:10 ]
[main] DEBUG MyTest - Skip class org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener (unchanged)
[main] TRACE MyTest - Constant [   9 ] [               12 ] [ NameAndTypeInfo:11:12 ]
[main] TRACE MyTest - Skip name-and-type ()V (unchanged)
[main] TRACE MyTest - Constant [  10 ] [                1 ] [ org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener (unchanged)
[main] TRACE MyTest - Skip UTF8 org/eclipse/jetty/servlets/DoSFilter$DoSTimeoutAsyncListener (unchanged)
[main] TRACE MyTest - Constant [  11 ] [                1 ] [ <init> ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: <init> (unchanged)
[main] TRACE MyTest - Skip UTF8 <init> (unchanged)
[main] TRACE MyTest - Constant [  12 ] [                1 ] [ ()V ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: ()V (unchanged)
[main] TRACE MyTest - Skip UTF8 ()V (unchanged)
[main] TRACE MyTest - Constant [  13 ] [                9 ] [ FieldrefInfo:14:15 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [  14 ] [                7 ] [ ClassInfo:16 ]
[main] DEBUG MyTest - Skip class org/eclipse/jetty/servlets/DoSFilter (unchanged)
[main] TRACE MyTest - Constant [  15 ] [               12 ] [ NameAndTypeInfo:17:18 ]
[main] TRACE MyTest - Skip name-and-type Ljava/util/Queue; (unchanged)
[main] TRACE MyTest - Constant [  16 ] [                1 ] [ org/eclipse/jetty/servlets/DoSFilter ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: org/eclipse/jetty/servlets/DoSFilter (unchanged)
[main] TRACE MyTest - Skip UTF8 org/eclipse/jetty/servlets/DoSFilter (unchanged)
[main] TRACE MyTest - Constant [  17 ] [                1 ] [ _queue ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: _queue (unchanged)
[main] TRACE MyTest - Skip UTF8 _queue (unchanged)
[main] TRACE MyTest - Constant [  18 ] [                1 ] [ Ljava/util/Queue; ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Ljava/util/Queue; (unchanged)
[main] TRACE MyTest - Skip UTF8 Ljava/util/Queue; (unchanged)
[main] TRACE MyTest - Constant [  19 ] [               10 ] [ MethodrefInfo:20:21 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [  20 ] [                7 ] [ ClassInfo:22 ]
[main] DEBUG MyTest - Class Reference: javax/servlet/AsyncEvent -> jakarta/servlet/AsyncEvent
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ Class Reference ] [ javax/servlet/AsyncEvent ] [ jakarta/servlet/AsyncEvent ]
[main] TRACE MyTest - Constant [  21 ] [               12 ] [ NameAndTypeInfo:23:24 ]
[main] DEBUG MyTest - NameAndType: ()Ljavax/servlet/AsyncContext; -> ()Ljakarta/servlet/AsyncContext;
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ NameAndType ] [ ()Ljavax/servlet/AsyncContext; ] [ ()Ljakarta/servlet/AsyncContext; ]
[main] TRACE MyTest - Constant [  22 ] [                1 ] [ javax/servlet/AsyncEvent ]
[main] TRACE MyTest - Input [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8Constant ] Initial [ javax/servlet/AsyncEvent ] Final [ jakarta/servlet/AsyncEvent ] ( [binary type] )
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: javax/servlet/AsyncEvent -> jakarta/servlet/AsyncEvent ([binary type])
[main] DEBUG MyTest - UTF8: javax/servlet/AsyncEvent -> jakarta/servlet/AsyncEvent
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8 ] [ javax/servlet/AsyncEvent ] [ jakarta/servlet/AsyncEvent ]
[main] TRACE MyTest - Constant [  23 ] [                1 ] [ getAsyncContext ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: getAsyncContext (unchanged)
[main] TRACE MyTest - Skip UTF8 getAsyncContext (unchanged)
[main] TRACE MyTest - Constant [  24 ] [                1 ] [ ()Ljavax/servlet/AsyncContext; ]
[main] TRACE MyTest - Input [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8Constant ] Initial [ ()Ljavax/servlet/AsyncContext; ] Final [ ()Ljakarta/servlet/AsyncContext; ] ( [binary descriptor] )
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: ()Ljavax/servlet/AsyncContext; -> ()Ljakarta/servlet/AsyncContext; ([binary descriptor])
[main] DEBUG MyTest - UTF8: ()Ljavax/servlet/AsyncContext; -> ()Ljakarta/servlet/AsyncContext;
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8 ] [ ()Ljavax/servlet/AsyncContext; ] [ ()Ljakarta/servlet/AsyncContext; ]
[main] TRACE MyTest - Constant [  25 ] [               11 ] [ InterfaceMethodrefInfo:26:27 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [  26 ] [                7 ] [ ClassInfo:28 ]
[main] DEBUG MyTest - Skip class java/util/Queue (unchanged)
[main] TRACE MyTest - Constant [  27 ] [               12 ] [ NameAndTypeInfo:29:30 ]
[main] TRACE MyTest - Skip name-and-type (Ljava/lang/Object;)Z (unchanged)
[main] TRACE MyTest - Constant [  28 ] [                1 ] [ java/util/Queue ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: java/util/Queue (unchanged)
[main] TRACE MyTest - Skip UTF8 java/util/Queue (unchanged)
[main] TRACE MyTest - Constant [  29 ] [                1 ] [ remove ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: remove (unchanged)
[main] TRACE MyTest - Skip UTF8 remove (unchanged)
[main] TRACE MyTest - Constant [  30 ] [                1 ] [ (Ljava/lang/Object;)Z ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: (Ljava/lang/Object;)Z (unchanged)
[main] TRACE MyTest - Skip UTF8 (Ljava/lang/Object;)Z (unchanged)
[main] TRACE MyTest - Constant [  31 ] [               10 ] [ MethodrefInfo:8:32 ]
[main] TRACE MyTest - Skip other (ignored)
[main] TRACE MyTest - Constant [  32 ] [               12 ] [ NameAndTypeInfo:33:34 ]
[main] DEBUG MyTest - NameAndType: (Ljavax/servlet/AsyncEvent;)V -> (Ljakarta/servlet/AsyncEvent;)V
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ NameAndType ] [ (Ljavax/servlet/AsyncEvent;)V ] [ (Ljakarta/servlet/AsyncEvent;)V ]
[main] TRACE MyTest - Constant [  33 ] [                1 ] [ onTimeout ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: onTimeout (unchanged)
[main] TRACE MyTest - Skip UTF8 onTimeout (unchanged)
[main] TRACE MyTest - Constant [  34 ] [                1 ] [ (Ljavax/servlet/AsyncEvent;)V ]
[main] TRACE MyTest - Input [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8Constant ] Initial [ (Ljavax/servlet/AsyncEvent;)V ] Final [ (Ljakarta/servlet/AsyncEvent;)V ] ( [binary descriptor] )
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: (Ljavax/servlet/AsyncEvent;)V -> (Ljakarta/servlet/AsyncEvent;)V ([binary descriptor])
[main] DEBUG MyTest - UTF8: (Ljavax/servlet/AsyncEvent;)V -> (Ljakarta/servlet/AsyncEvent;)V
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8 ] [ (Ljavax/servlet/AsyncEvent;)V ] [ (Ljakarta/servlet/AsyncEvent;)V ]
[main] TRACE MyTest - Constant [  35 ] [                1 ] [ (Lorg/eclipse/jetty/servlets/DoSFilter;)V ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: (Lorg/eclipse/jetty/servlets/DoSFilter;)V (unchanged)
[main] TRACE MyTest - Skip UTF8 (Lorg/eclipse/jetty/servlets/DoSFilter;)V (unchanged)
[main] TRACE MyTest - Constant [  36 ] [                1 ] [ Code ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Code (unchanged)
[main] TRACE MyTest - Skip UTF8 Code (unchanged)
[main] TRACE MyTest - Constant [  37 ] [                1 ] [ LineNumberTable ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: LineNumberTable (unchanged)
[main] TRACE MyTest - Skip UTF8 LineNumberTable (unchanged)
[main] TRACE MyTest - Constant [  38 ] [                1 ] [ LocalVariableTable ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: LocalVariableTable (unchanged)
[main] TRACE MyTest - Skip UTF8 LocalVariableTable (unchanged)
[main] TRACE MyTest - Constant [  39 ] [                1 ] [ this ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: this (unchanged)
[main] TRACE MyTest - Skip UTF8 this (unchanged)
[main] TRACE MyTest - Constant [  40 ] [                1 ] [ Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener; ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener; (unchanged)
[main] TRACE MyTest - Skip UTF8 Lorg/eclipse/jetty/servlets/DoSFilter$DoSAsyncListener; (unchanged)
[main] TRACE MyTest - Constant [  41 ] [                1 ] [ MethodParameters ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: MethodParameters (unchanged)
[main] TRACE MyTest - Skip UTF8 MethodParameters (unchanged)
[main] TRACE MyTest - Constant [  42 ] [                1 ] [ event ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: event (unchanged)
[main] TRACE MyTest - Skip UTF8 event (unchanged)
[main] TRACE MyTest - Constant [  43 ] [                1 ] [ Ljavax/servlet/AsyncEvent; ]
[main] TRACE MyTest - Input [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8Constant ] Initial [ Ljavax/servlet/AsyncEvent; ] Final [ Ljakarta/servlet/AsyncEvent; ] ( [binary descriptor] )
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Ljavax/servlet/AsyncEvent; -> Ljakarta/servlet/AsyncEvent; ([binary descriptor])
[main] DEBUG MyTest - UTF8: Ljavax/servlet/AsyncEvent; -> Ljakarta/servlet/AsyncEvent;
[main] TRACE MyTest - Class [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] [ UTF8 ] [ Ljavax/servlet/AsyncEvent; ] [ Ljakarta/servlet/AsyncEvent; ]
[main] TRACE MyTest - Constant [  44 ] [                1 ] [ Exceptions ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: Exceptions (unchanged)
[main] TRACE MyTest - Skip UTF8 Exceptions (unchanged)
[main] TRACE MyTest - Constant [  45 ] [                7 ] [ ClassInfo:46 ]
[main] DEBUG MyTest - Skip class java/io/IOException (unchanged)
[main] TRACE MyTest - Constant [  46 ] [                1 ] [ java/io/IOException ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: java/io/IOException (unchanged)
[main] TRACE MyTest - Skip UTF8 java/io/IOException (unchanged)
[main] TRACE MyTest - Constant [  47 ] [                1 ] [ SourceFile ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: SourceFile (unchanged)
[main] TRACE MyTest - Skip UTF8 SourceFile (unchanged)
[main] TRACE MyTest - Constant [  48 ] [                1 ] [ DoSFilter.java ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: DoSFilter.java (unchanged)
[main] TRACE MyTest - Skip UTF8 DoSFilter.java (unchanged)
[main] TRACE MyTest - Constant [  49 ] [                1 ] [ NestHost ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: NestHost (unchanged)
[main] TRACE MyTest - Skip UTF8 NestHost (unchanged)
[main] TRACE MyTest - Constant [  50 ] [                1 ] [ InnerClasses ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: InnerClasses (unchanged)
[main] TRACE MyTest - Skip UTF8 InnerClasses (unchanged)
[main] TRACE MyTest - Constant [  51 ] [                1 ] [ DoSAsyncListener ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: DoSAsyncListener (unchanged)
[main] TRACE MyTest - Skip UTF8 DoSAsyncListener (unchanged)
[main] TRACE MyTest - Constant [  52 ] [                1 ] [ DoSTimeoutAsyncListener ]
[main] TRACE MyTest -     String org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener UTF8Constant: DoSTimeoutAsyncListener (unchanged)
[main] TRACE MyTest - Skip UTF8 DoSTimeoutAsyncListener (unchanged)
[main] DEBUG MyTest - Stop processing [ org.eclipse.jetty.servlets.DoSFilter$DoSAsyncListener ] using [ Class Action ]: Content changes
Exception in thread "main" java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at aQute.bnd.classfile.ConstantPool.utf8Info(ConstantPool.java:710)
	at aQute.bnd.classfile.MethodParametersAttribute$MethodParameter.write(MethodParametersAttribute.java:78)
	at aQute.bnd.classfile.MethodParametersAttribute.write(MethodParametersAttribute.java:44)
	at aQute.bnd.classfile.Attribute.writeAttributes(Attribute.java:134)
	at aQute.bnd.classfile.MemberInfo.write(MemberInfo.java:55)
	at aQute.bnd.classfile.ClassFile.write(ClassFile.java:134)
	at org.eclipse.transformer.action.impl.ClassActionImpl.apply(ClassActionImpl.java:488)
	at MyTest.transform(MyTest.java:52)
	at MyTest.main(MyTest.java:42)
```
[eclipse-transformer-npe.zip](https://github.com/eclipse/transformer/files/14615028/eclipse-transformer-npe.zip)
[eclipse-transformer-npe.jar.zip](https://github.com/eclipse/transformer/files/14614976/eclipse-transformer-npe.jar.zip)

**Comment by bjhargrave** [comment]
This is a bug in Bnd that was fixed for Bnd 7.0.0. See https://github.com/bndtools/bnd/pull/5690.

This was reported previously. https://github.com/eclipse/transformer/issues?q=is%3Aissue+7.0.0+is%3Aclosed

See https://github.com/eclipse/transformer/tree/main/maven-plugins/transformer-maven-plugin#java-21-or-later which describes how to configure your pom to use Bnd 7.0.0.

**Comment by jluehe** [comment]
Thank you, @bjhargrave, for the quick response! Upgrading to `Bnd 7.0.0` did indeed fix the issue.
